### DISCUSSION TITLE
I2C and timer

### DISCUSSION THREAD

**Discussion by giantsmoon** [micropython/micropython#13175] [ESP32]
Hi there!
i start to write a program for my ESP32 and i comunicate with a ADC on I2C bus and a OLED and everythingh work perfect . Now i use a timer configurate in periodic time for read the ADC but the data i receive are all corrupt and i don't understand why . I post the code and i hope some one can hle me 
Thank you so much 
Sergio 

```py
from machine import  Pin, SoftI2C
from machine import Timer

import ssd1327
import mcp3424


i2c = SoftI2C(scl=Pin(22), sda=Pin(21), freq = 400000)
add = 0x68

display = ssd1327.SEEED_OLED_96X96(i2c)
timer_0 = Timer(1)


def interruption_handler(timer):
    value= test.one_shot()
    tens = ((2048/2**17)* value)/1000
    print(tens)
   
timer_0.init(mode=Timer.PERIODIC, period=3000, callback=interruption_handler)
test= mcp3424.MCP34XX(1,18,1, add,i2c)
display.fill(0)
display.text('Hello World', 0,  0)
display.show()

#value= test.one_shot()
#tens = ((2048/2**17)* value)/1000
#print(tens)
```

(Edit by @jimmo)

**Comment by jimmo** [comment]
Please see [Welcome (Please read before posting)](https://github.com/orgs/micropython/discussions/9111) for how to format code snippets.

**Comment by jimmo** [comment]
Where does `mcp3424` come from?

FWIW, my guess is that you are potentially racing against the display driver which could be in the middle of a transaction on the same i2c bus?
